src/Agent/NewRelic/Agent/Core/Config/BootstrapConfiguration.cs

```diff
@@ -184,5 +192,145 @@ private ValueWithProvenance<bool> TryGetAgentEnabledSetting(Func<string, ValueWi
 
             return null;
         }
+
+        private class BootstrapLogConfig : ILogConfig
+        {
+            private readonly string _directoryFromLocalConfig;
+            private readonly string _fileNameFromLocalConfig;
+            private readonly string _logRollingStrategyFromLocalConfig;
+            private readonly IProcessStatic _processStatic;
+            private readonly Predicate<string> _checkDirectoryExists;
+            private readonly Func<string, string> _getFullPath;
+
+            public BootstrapLogConfig(configurationLog localLogConfiguration, IProcessStatic processStatic, Predicate<string> checkDirectoryExists, Func<string, string> getFullPath)
+            {
+                _processStatic = processStatic;
+                _checkDirectoryExists = checkDirectoryExists;
+                _getFullPath = getFullPath;
+
+                _directoryFromLocalConfig = localLogConfiguration.directory;
+                _fileNameFromLocalConfig = localLogConfiguration.fileName;
+                _logRollingStrategyFromLocalConfig = localLogConfiguration.logRollingStrategy.ToString();
+
+                Enabled = ConfigLoaderHelpers.GetLoggingEnabledValue(localLogConfiguration);
+                Console = ConfigLoaderHelpers.GetOverride("NEW_RELIC_LOG_CONSOLE", localLogConfiguration.console);
+                IsAuditLogEnabled = localLogConfiguration.auditLog;
+                MaxLogFileSizeMB = ConfigLoaderHelpers.GetOverride("NEW_RELIC_LOG_MAX_FILE_SIZE_MB", localLogConfiguration.maxLogFileSizeMB);
+                MaxLogFiles = ConfigLoaderHelpers.GetOverride("NEW_RELIC_LOG_MAX_FILES", localLogConfiguration.maxLogFiles);
+
+                // The log level needs to be initialized after the Enabled property is initialized because it depends on that value
+                LogLevel = ConfigLoaderHelpers.GetLoggingLevelValue(localLogConfiguration, Enabled);
+
+                _checkDirectoryExists = checkDirectoryExists;
+                _getFullPath = getFullPath;
+            }
+
+            public bool Enabled { get; }
+
+            public string LogLevel { get; }
+
+            public bool Console { get; }
+
+            public bool IsAuditLogEnabled { get; }
+
+            public int MaxLogFileSizeMB { get; }
+
+            public int MaxLogFiles { get; }
+
+            private LogRollingStrategy? _logRollingStrategy;
+            public LogRollingStrategy LogRollingStrategy
+            {
+                get
+                {
+                    return _logRollingStrategy ?? (_logRollingStrategy = GetLogRollingStrategy()).Value;
+                }
+            }
+
+            public string GetFullLogFileName()
```

Comment:
I know this logic hasn't changed, but I think we could simplify it a bit by using `Path.Combine`.